### PR TITLE
Uniformly check returned error on OK response in backend

### PR DESCRIPTION
**Pull request by artek-koltun**
see https://github.com/opiproject/opi-spdk-bridge/issues/501

### CODE REVIEW

File: pkg/backend/aio_test.go [hunk 1]
```diff
@@ -360,12 +360,12 @@ func TestBackEnd_ListAioControllers(t *testing.T) {
 		size    int32
 		token   string
 	}{
-		"valid request with invalid SPDK response": {
+		"valid request with empty result SPDK response": {
 			testAioVolumeID,
 			nil,
 			[]string{`{"id":%d,"error":{"code":0,"message":""},"result":[]}`},
-			codes.InvalidArgument,
-			fmt.Sprintf("Could not find any namespaces for NQN: %v", "nqn.2022-09.io.spdk:opi3"),
```
Comment (glimchb) [inline pkg/backend/aio_test.go:368]: how this passed before ?

**Review by glimchb — APPROVED** [review]
nice consolidation

File: pkg/backend/aio_test.go (hunk 1 above)
Comment (artek-koltun) [reply pkg/backend/aio_test.go:368]: server returned OK status. And we checked against expectations only if returned error is not OK

Comment (artek-koltun) [reply pkg/backend/aio_test.go:368]: like here
https://github.com/opiproject/opi-spdk-bridge/blob/c2247af22c347f22af89845483342137e15d9c2c/pkg/backend/aio_test.go#L144

Comment (glimchb) [reply pkg/backend/aio_test.go:368]: understood

**Pull request merged**